### PR TITLE
Exposing step ca to the internet

### PR DESCRIPTION
**Pull request by tashian**

### CODE REVIEW

File: step-ca/certificate-authority-server-production.mdx [hunk 1]
```diff
@@ -502,13 +506,72 @@ Layer 7 proxying is _not recommended_, becase the `step` toolchain is built arou
...
+- **Enumeration of CA provisioners**  
+  The `/provisioners` exposes provisioner configuration and is included for client convenience. It includes JWK encrypted private keys, OIDC endpoint URLs, and more. Without it, clients will need to pass more configuration flags into `step ca certificate`.
+
+### Run A Reverse Proxy
+
+Rather than expose your entire CA to the internet,
+use a reverse proxy to expose only the necessary CA endpoints:
+- `/root/*` will allow clients to bootstrap with `step ca bootstrap`
+- `/renew` for mutual TLS or token-based renewals
+- `/1.0/sign` for JWK certificate issuance
```
Comment (maraino) [inline step-ca/certificate-authority-server-production.mdx:528]: If `/provisioners` is not exposed, you need the private key to sign a JWT.

File: step-ca/certificate-authority-server-production.mdx [hunk 2]
```diff
@@ -502,13 +506,72 @@ Layer 7 proxying is _not recommended_, becase the `step` toolchain is built arou
...
+    By default, ACME provisioners will issue certificates to <b>any client</b> that can satisfy an ACME challenge.
+    Do not expose an ACME provisioner to the internet without precautions.
+    This warning applies to all challenge types, including ACME `device-attest-01`.
+  </div>
+</Alert>
+
+When connecting an ACME provisioner to the internet, you will need to take precautions:
+- Use explicit [policies](./policies.mdx) to limit which subject names can request certificates via ACME.
+- Hide your ACME provisioner  
+  1. Use an unguessable shared secret as your ACME provisioner name:
```
Comment (maraino) [inline step-ca/certificate-authority-server-production.mdx]: Only if `/provisioners` is not available.

**Review by maraino — COMMENTED** [review]
Looks good. I've just made a couple of comments.

File: step-ca/certificate-authority-server-production.mdx [hunk 3]
```diff
@@ -502,13 +506,73 @@ Layer 7 proxying is _not recommended_, becase the `step` toolchain is built arou
...
+With some precautions,
+`step-ca` may be exposed to the public internet.
+
+If you choose to expose a CA to the internet, consider these potential risks:
+- **Exposed, unauthenticated ACME provisioners**  
+  By default, any ACME client that can reach the CA and satisfy an ACME challenge can enroll in your PKI.
+- **Enumeration of SSH infrastructure**  
+  The `/ssh/hosts` endpoint lists all enrolled hosts.
+- **Enumeration of CA provisioners**  
+  The `/provisioners` exposes provisioner configuration and is included for client convenience. It includes JWK encrypted private keys, OIDC endpoint URLs, and more. Without it, clients will need to pass more configuration flags into `step ca certificate`.
```
Comment (maraino) [inline step-ca/certificate-authority-server-production.mdx]: OIDC client credentials and endpoint URLs.

**Review by andysmallstep — APPROVED** [review]

**Pull request merged**